base/file.jl

```diff
@@ -341,37 +354,47 @@ Obtain the path of a temporary directory (possibly shared with other processes).
 tempdir()
 
 """
-    tempname()
+    tempname(; prefix="$temp_prefix")
+
+Generate a unique temporary file path. A `prefix` can be provided in addition, which
+otherwise defaults to "$temp_prefix".
 
-Generate a unique temporary file path.
+*Note: The `prefix` may be truncated based on the limitations of the OS.*
```

Comment:
AFAICT we won't be able to get rid of the 3-char limit on Windows even for `mktemp`, so the issue is more general than `tempname`. Anyway my remark applied to other docstrings too.